projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -195,8 +208,9 @@ private[projection] class R2dbcOffsetStore(
     VALUES (?,?,?,?,?,?, transaction_timestamp())"""
 
   // delete less than a timestamp
-  private val deleteOldTimestampOffsetSql: String =
-    sql"DELETE FROM $timestampOffsetTable WHERE slice BETWEEN ? AND ? AND projection_name = ? AND timestamp_offset < ?"
+  private val deleteOldTimestampOffsetSql: String = sql"""
+    DELETE FROM $timestampOffsetTable WHERE slice BETWEEN ? AND ? AND projection_name = ? AND timestamp_offset < ?
+    AND NOT (slice || '-' || persistence_id || '-' || seq_nr) = ANY (?)"""
```

Comment:
Second thought, the slice is probably not needed here since it's anyway string concatenation. I'll take a look at explain analyze.